ocaml/xs-trace/test/test_xs_trace.ml

```diff
@@ -0,0 +1,29 @@
+let _ =
+  let inet_addr = Unix.ADDR_INET (Unix.inet_addr_of_string "127.0.0.1", 9411) in
+  let read_body ic file len =
+    let line = really_input_string ic !len in
+    Printf.fprintf file "%s\n" line
+  in
+  let len = ref 0 in
+  let rec read_header ic file =
```

Comment:
I wanted to suggest using the library xapi has to implement  http server, but it's a mess :(